engine/gamesys/proto/gamesys/model_ddf.proto

```diff
@@ -9,24 +9,40 @@ option java_outer_classname = "ModelProto";
 
 // TODO: Add documentation for messages, see https://github.com/defold/extension-spine/blob/main/defold-spine/commonsrc/spine_ddf.proto
 
+message Texture
+{
+    required string sampler     = 1; // the name of the sampler
+    required string texture     = 2 [(resource)=true];
+}
+
+message Material
+{
+    required string name        = 1; // the name of the material in the model file
+    required string material    = 2 [(resource)=true];
+    repeated Texture textures   = 3;
+}
+
+// The source format (.model)
 message ModelDesc
 {
     required string mesh        = 2 [(resource)=true];
-    required string material    = 3 [(resource)=true];
-    repeated string textures    = 4 [(resource)=true];
+    required string material    = 3 [(resource)=true]; // Deprecated
+    repeated string textures    = 4 [(resource)=true]; // Deprecated
     optional string skeleton    = 5 [(resource)=true];
     optional string animations  = 6 [(resource)=true];
     optional string default_animation  = 7;
 
     optional string name        = 10; // Deprecated
+    repeated Material materials = 11;
```

Comment:
Should we make the deprecated `material` field `optional` now? I guess we'll need to migrate the format in the editor, but after that it would be nice to not have to specify this old field anymore?